### PR TITLE
write dismod output to logs in real time

### PR DESCRIPTION
**Pull request by alec-deason**

### CODE REVIEW

File: src/cascade/model/session.py [hunk 1]
```diff
@@ -349,3 +345,62 @@ def simulation(self, index):
             the Nth simulation.
         """
         return self._file_objects.read_simulation_model_and_data(self._model, self._data, index)
+
```
Comment (adolgert) [inline src/cascade/model/session.py]: These functions are of a piece, and they might make sense in their own file?

File: src/cascade/model/session.py [hunk 2]
```diff
@@ -349,3 +345,62 @@ def simulation(self, index):
...
+        std_out_task = loop.create_task(_read_pipe(sub_process.stdout, dismod_report_info(stdout_accumulator)))
+        std_err_task = loop.create_task(_read_pipe(sub_process.stderr, dismod_report_stderr(stderr_accumulator)))
+
+        await sub_process.wait()
+        await std_out_task
+        await std_err_task
+
+        return sub_process.returncode, "".join(stdout_accumulator), "".join(stderr_accumulator)
+
+    return loop.run_until_complete(coroutine())
```
Comment (adolgert) [inline src/cascade/model/session.py]: That all looked very familiar and has been working.

File: tests/model/test_session.py [hunk 3]
```diff
@@ -42,3 +44,27 @@ def test_options(dismod):
     ))
     # Run a predict in order to verify the options are accepted.
     session.predict(model_var, avgints, parent_location)
+
```
Comment (adolgert) [inline tests/model/test_session.py]: I like the bash test and /bin/false. I don't know what will happen when the run command fails b/c it isn't found. There's no error-condition testing. This is how it was, so I'm just commenting.

Comment (alec-deason) [reply tests/model/test_session.py]: It raises a FileNotFound exception. We could wrap it in a custom exception but it's really not a recoverable error the way this tool gets used so I think it's a fine choice to just let it bubble out.

File: src/cascade/model/session.py (hunk 1 above)
Comment (alec-deason) [reply src/cascade/model/session.py]: Bleh. I'm struggling with which package that file would be in. It clearly doesn't belong in model. It seems like it belongs in executor but I think you said you didn't want other packages to have dependencies on executor, which is totally reasonable. Ok I'm going to make `cascade.core.subprocess_utils` That's not perfect but I think it's good enough and if we want to move it later, that's easy.

**Pull request merged**